service/app.py

```diff
@@ -136,6 +136,53 @@ def get_professor_data():
     # Include the full instructor name in the response
     return jsonify({"professor": full_instructor_name, "courses": course_data})
 
+@app.route('/class', methods=['GET'])
+def get_class_data():
+    class_id = request.args.get('classId') 
+    if not class_id:
+        return jsonify({"error": "Class ID is required"}), 400
+
+    # Fetch class data from the database
+    class_data = ClassData.query.filter_by(class_name=class_id).first() 
+
+    if class_data:
+        return jsonify({"class": {
+            "title": f"{class_data.subject} {class_data.class_name}",
+            "code": class_data.class_name,
```

Comment:
I think "code" should be class_data.class_nbr assuming you're trying to get the 4/5 digit class ID that you could use to search the class in Louie etc.

Reply:
This affects the grade distribution graph so I'm going to leave it as it is for now, but in the future I'll add the digit class ID for search functionality.